crates/binjs_generate_library/src/lib.rs

```diff
@@ -1511,8 +1516,18 @@ pub trait Visitor<E, G=()> where G: for<'me> From<&'me Self> {{
 }}\n
 pub trait Walker<'a>: Sized {{
     type Output;
-    fn walk<V, E, G>(&'a mut self, path: &mut WalkPath, visitor: &mut V) -> Result<Option<Self::Output>, E> where V: Visitor<E, G>, G: for<'w> From<&'w V>;
+    fn walk<V, E, G>(&'a mut self, path: &mut WalkPath, visitor: &mut V) -> Result<Option<Self::Output>, E> where V: Visitor<E, G>, G: WalkGuard<V>;
 }}\n
+pub trait WalkGuard<V> where Self: Sized {{
+    fn new(visitor: &V, path: &WalkPath) -> Self;
+}}\n
```

Comment:
just noticed
why is this using escape sequence instead of raw newline?

Reply:
Oops, fixed.